update/sync/syncTries.go

```diff
@@ -0,0 +1,150 @@
+package sync
+
+import (
+	"bytes"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/state/factory"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/update"
+)
+
+type syncTries struct {
+	tries       map[string]data.Trie
+	trieSyncers update.TrieSyncContainer
+	activeTries state.TriesHolder
+}
+
+type ArgsNewSyncTriesHandler struct {
+	TrieSyncers update.TrieSyncContainer
+	ActiveTries state.TriesHolder
+}
+
+func NewSyncTriesHandler(args ArgsNewSyncTriesHandler) (*syncTries, error) {
+	if check.IfNil(args.TrieSyncers) {
+		return nil, update.ErrNilTrieSyncers
+	}
+	if check.IfNil(args.ActiveTries) {
+		return nil, update.ErrNilActiveTries
+	}
+
+	st := &syncTries{
+		tries:       make(map[string]data.Trie),
+		trieSyncers: args.TrieSyncers,
+		activeTries: args.ActiveTries,
+	}
+
+	return st, nil
+}
+
+func (st *syncTries) SyncTriesFrom(meta *block.MetaBlock, wg *sync.WaitGroup) error {
+	var errFound error
+	mutErr := sync.Mutex{}
+
+	go func() {
+		errMeta := st.syncMeta(meta, wg)
+		if errMeta != nil {
+			mutErr.Lock()
+			errFound = errMeta
+			mutErr.Unlock()
+		}
```

Comment:
`wg.Done` should be moved here

Reply:
done.

Reply:
I think you forgot to remove the `wg.Done` from `syncMeta` and `syncShard`

Reply:
done.